File: src/main/java/web/fidex/model/fidex_model/Product.java
package web.fidex.model.fidex_model;

import java.io.Serializable;
import java.util.Objects;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table(name = "product")
public class Product implements Serializable {

	private static final long serialVersionUID = 1L;

	@Id
	private Long id;

	@NotBlank(message = "O nome do produto é obrigatório")
	private String name;
	@Min(value = 0)
	private Double price;
	@Min(value = 0)
	private Long quantity;
	private Status status = Status.ATIVO;
	private String createdBy;

	public String getCreatedBy() {
		return createdBy;
	}

	public void setCreatedBy(String createdBy) {
		this.createdBy = createdBy;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getQuantity() {
		return quantity;
	}

	public void setQuantity(Long quantity) {
		this.quantity = quantity;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public Double getPrice() {
		return price;
	}

	public void setPrice(Double price) {
		this.price = price;
	}

	public int getPoints() {
		if (price == null) {
			return 0;
		}
		return (int) (price * 1);
	}

	public Status getStatus() {
		return status;
	}

	public void setStatus(Status status) {
		this.status = status;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Product other = (Product) obj;
		return Objects.equals(id, other.id);
	}

	public static long getSerialversionuid() {
		return serialVersionUID;
	}

}

File: src/main/kotlin/web/fidex/model/fidex_model/Product.kt
package web.fidex.model.fidex_model

import jakarta.persistence.*
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import java.io.Serializable
import java.util.*

@Entity
@Table(name = "product")
class Product(
        @Id
        @SequenceGenerator(name = "gerador3", sequenceName = "product_id_seq", allocationSize = 1)
        @GeneratedValue(generator = "gerador3", strategy = GenerationType.SEQUENCE)
        var id: Long? = null,
        @field:NotBlank(message = "O nome do produto é obrigatório") var name: String? = null,
        @field:Min(0) var price: Double? = null,
        @field:Min(0) var quantity: Long? = null,
        @Enumerated(EnumType.STRING) var status: Status = Status.ATIVO,
        var createdBy: String? = null
) : Serializable {

    val points: Int
        get() = (price ?: 0.0).toInt()

    override fun hashCode(): Int = Objects.hash(id)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this::class != other::class) return false
        val that = other as Product
        return id == that.id
    }

    companion object {
        private const val serialVersionUID = 1L

        @JvmStatic fun getSerialversionuid(): Long = serialVersionUID
    }
}
